api/v1beta1/infraenv_types.go

```diff
@@ -89,13 +89,13 @@ type InfraEnvSpec struct {
 	// +optional
 	CpuArchitecture string `json:"cpuArchitecture,omitempty"`
 
-	// IPXEScriptType the script type that should be served (HostRedirect/Boot)
-	// When the value is HostRedirect, the served script consists from URL redirection (i.e chain <url>).  The
-	// <url> is the same URL that contains query parameter mac=${net0/mac}.  The value is IPXE macro is replaced
-	//  with the mac address.  The mac address is used to identify the invoking script.
-	// When the value is Boot (or default), the served script contains a script that causes the invoker to boot
-	// the discovery ISO from the network.
-	// +kubebuilder:default=Boot
+	// IPXEScriptType the script type that should be served (DiscoveryImageAlways/BootOrderControl)
+	// DiscoveryImageAlways: Boot unconditionaly from the network discovery image
+	// BootOrderControl: Boot from discovery ISO depending on the host's state.
+	// When the value is BootOrderControl, the service will look for an Agent record that matches the host's MAC address;
+	// if found, and if that Host is in a state where disk image was already written to disk, then host will boot the host disk.
```

Comment:
> where disk image was already written to disk,
... and the Agent is still bound to a cluster,
?

Reply:
The condition is according to the installing stage the host is in - not if it is provisioned.  Only after disk image has been written to disk, the boot from discovery ISO is skipped.

Reply:
Ok. Maybe the part about being bound to a cluster will be added after we fully support this: https://github.com/openshift/assisted-service/pull/3849